### PR TITLE
Pick up shnupbup's stuff

### PR DESCRIPTION
**Pull request by liach**
Supersedes #1282 (shnupbups indicated loss of interest to maintain it)

Removed duplicated mapping and mapped a lot of details missed by shnupbups

### CODE REVIEW

File: mappings/net/minecraft/structure/BastionBridgeData.mapping [hunk 1]
```diff
@@ -0,0 +1,2 @@
+CLASS net/minecraft/class_5174 net/minecraft/structure/BastionBridgeData
+	METHOD method_27198 init ()V
```
Comment (YanisBft) [inline mappings/net/minecraft/structure/BastionBridgeData.mapping:2]: ~~`initialize`, to match the village data classes~~
i'm stupid

Comment (liach) [reply mappings/net/minecraft/structure/BastionBridgeData.mapping:2]: See #618, where we've decided to use `init` globally.

File: mappings/net/minecraft/block/Blocks.mapping [hunk 2]
```diff
@@ -7,6 +7,18 @@ CLASS net/minecraft/class_2246 net/minecraft/block/Blocks
 	METHOD method_26110 createShulkerBoxBlock (Lnet/minecraft/class_1767;Lnet/minecraft/class_4970$class_2251;)Lnet/minecraft/class_2480;
 		ARG 0 color
 		ARG 1 settings
+	METHOD method_26113 allow (Lnet/minecraft/class_2680;Lnet/minecraft/class_1922;Lnet/minecraft/class_2338;)Z
+		COMMENT A shortcut to allow in a context predicate, used as {@code Blocks::allow}.
+		ARG 0 state
+		ARG 1 view
```
Comment (YanisBft) [inline mappings/net/minecraft/block/Blocks.mapping]:
```suggestion
		ARG 1 world
```

File: mappings/net/minecraft/block/Blocks.mapping [hunk 3]
```diff
@@ -7,6 +7,18 @@ CLASS net/minecraft/class_2246 net/minecraft/block/Blocks
...
+		ARG 2 pos
+	METHOD method_26114 refuse (Lnet/minecraft/class_2680;Lnet/minecraft/class_1922;Lnet/minecraft/class_2338;Lnet/minecraft/class_1299;)Ljava/lang/Boolean;
+		COMMENT A shortcut to refuse in a typed context predicate with an {@link EntityType},
+		COMMENT used as {@code Blocks::refuse}.
+		ARG 0 state
+		ARG 1 view
```
Comment (YanisBft) [inline mappings/net/minecraft/block/Blocks.mapping]:
```suggestion
		ARG 1 world
```

File: mappings/net/minecraft/block/Blocks.mapping [hunk 4]
```diff
@@ -16,11 +28,24 @@ CLASS net/minecraft/class_2246 net/minecraft/block/Blocks
 		ARG 0 sticky
 	METHOD method_26120 createStainedGlassBlock (Lnet/minecraft/class_1767;)Lnet/minecraft/class_2506;
 		ARG 0 color
+	METHOD method_26122 refuse (Lnet/minecraft/class_2680;Lnet/minecraft/class_1922;Lnet/minecraft/class_2338;)Z
+		COMMENT A shortcut to refuse in a context predicate, used as {@code Blocks::refuse}.
+		ARG 0 state
+		ARG 1 view
```
Comment (YanisBft) [inline mappings/net/minecraft/block/Blocks.mapping]:
```suggestion
		ARG 1 world
```

File: mappings/net/minecraft/block/Blocks.mapping [hunk 5]
```diff
@@ -16,11 +28,24 @@ CLASS net/minecraft/class_2246 net/minecraft/block/Blocks
...
+		ARG 2 pos
+	METHOD method_26123 allow (Lnet/minecraft/class_2680;Lnet/minecraft/class_1922;Lnet/minecraft/class_2338;Lnet/minecraft/class_1299;)Ljava/lang/Boolean;
+		COMMENT A shortcut to allow in a typed context predicate with an {@link EntityType},
+		COMMENT used as {@code Blocks::allow}.
+		ARG 0 state
+		ARG 1 view
```
Comment (YanisBft) [inline mappings/net/minecraft/block/Blocks.mapping]:
```suggestion
		ARG 1 world
```

File: mappings/net/minecraft/world/gen/feature/RuinedPortalFeaturePiece.mapping [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+		FIELD field_24023 cold Z
+		FIELD field_24024 mossiness F
+		FIELD field_24025 airPocket Z
+		FIELD field_24026 overgrown Z
+		FIELD field_24027 vines Z
+		FIELD field_24028 replaceWithBlackstone Z
+		METHOD <init> (Lcom/mojang/datafixers/Dynamic;)V
+			ARG 1 dynamic
+		METHOD method_27251 serialize (Lcom/mojang/datafixers/types/DynamicOps;)Ljava/lang/Object;
+			ARG 1 opes
```
Comment (YanisBft) [inline mappings/net/minecraft/world/gen/feature/RuinedPortalFeaturePiece.mapping]:
```suggestion
			ARG 1 ops
```

**Review by YanisBft — APPROVED** [review]

File: mappings/net/minecraft/block/Blocks.mapping [hunk 7]
```diff
@@ -7,6 +7,18 @@ CLASS net/minecraft/class_2246 net/minecraft/block/Blocks
 	METHOD method_26110 createShulkerBoxBlock (Lnet/minecraft/class_1767;Lnet/minecraft/class_4970$class_2251;)Lnet/minecraft/class_2480;
 		ARG 0 color
 		ARG 1 settings
+	METHOD method_26113 allow (Lnet/minecraft/class_2680;Lnet/minecraft/class_1922;Lnet/minecraft/class_2338;)Z
+		COMMENT A shortcut to allow in a context predicate, used as {@code Blocks::allow}.
```
Comment (Runemoro) [inline mappings/net/minecraft/block/Blocks.mapping]: This comment is a bit confusing... It should describe what the method does. What does it allow?

Comment (liach) [reply mappings/net/minecraft/block/Blocks.mapping]: It is passed to settings, like `blockVision(Blocks::allow)`

Comment (liach) [reply mappings/net/minecraft/block/Blocks.mapping]: Hmm, should prob rename them to always/never

Comment (Runemoro) [reply mappings/net/minecraft/block/Blocks.mapping]: I see, so these just always return `true`/`false`?

Comment (liach) [reply mappings/net/minecraft/block/Blocks.mapping]: Yep.

Comment (Runemoro) [reply mappings/net/minecraft/block/Blocks.mapping]: Yeah, `always` and `never` would be better. `allow` sounds more like a method that adds something to a list of allowed things, than just returning true.

**Review by Runemoro — APPROVED** [review]

**Pull request merged**